src/coreclr/src/vm/diagnosticserver.h

```diff
@@ -8,45 +8,46 @@
 #ifdef FEATURE_PERFTRACING // This macro should change to something more generic than performance.
 
 #include "diagnosticsipc.h"
-
-enum class DiagnosticMessageType : uint32_t
-{
-    ///////////////////////////////////////////////////////////////////////////
-    // Debug = 0
-    GenerateCoreDump = 1,           // Initiates core dump generation
-
-    ///////////////////////////////////////////////////////////////////////////
-    // EventPipe = 1024
-    StartEventPipeTracing = 1024,   // To file
-    StopEventPipeTracing,
-    CollectEventPipeTracing,        // To IPC
-
-    ///////////////////////////////////////////////////////////////////////////
-    // Profiler = 2048
-    AttachProfiler = 2048,
-};
-
-struct MessageHeader
-{
-    DiagnosticMessageType RequestType;
-    uint32_t Pid;
-};
+#include "diagnosticsprotocol.h"
 
 //! Defines an implementation of a IPC handler that dispatches messages to the runtime.
 class DiagnosticServer final
 {
 public:
-    //! Initialize the event pipe (Creates the EventPipe IPC server).
+    // Initialize the event pipe (Creates the EventPipe IPC server).
     static bool Initialize();
 
-    //! Shutdown the event pipe.
+    // Shutdown the event pipe.
     static bool Shutdown();
 
-    //! Diagnostics server thread.
+    // Diagnostics server thread.
     static DWORD WINAPI DiagnosticsServerThread(LPVOID lpThreadParameter);
 
+    // Pauses EEStartup after the Diagnostics Server has been started
+    // allowing a Diagnostics Monitor to attach perform tasks before
+    // Startup is completed
+    NOINLINE static void PauseForDiagnosticsMonitor();
+
+    // Sets CLREvent to resume startup in EEStartupHelper (see: ceemain.cpp:686 for pausing point)
```

Comment:
nit: If someone changes ceemain.cpp, the line number in this comment might get outdated real soon. Can we describe it more like: (after the call to `PAL_SetShutdownCallback` in `EEStartupHelper`)?